src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.ts

```diff
@@ -267,15 +266,14 @@ export class FileUploadAssessmentComponent implements OnInit, OnDestroy {
         this.unreferencedFeedback = [];
         this.fileUploadSubmissionService.getSubmissionWithoutAssessment(this.exercise!.id!, false, this.correctionRound).subscribe({
             next: (submission: FileUploadSubmission) => {
+                this.isLoading = false;
+                this.unassessedSubmission = submission;
                 if (!submission) {
-                    // there are no unassessed submission, nothing we have to worry about
-                    this.hasNewSubmissions = false;
+                    // there are no unassessed submissions
+                    this.submission = submission;
```

Comment:
I think this is more clear:
```suggestion
                    this.submission = undefined;
```

This also reveals that `submission` has the wrong type: It's not `undefined`-able, but actually, its values can be `undefined`.